File: Python/generatePascal.py
class Solution:
    # @param A : integer
    # @return a list of list of integers
    def generate(self, numRows):
        result = []
        for i in xrange(numRows):
            result.append([])
            for j in xrange(i + 1):
                if j in (0, i):
                    result[i].append(1)
                else:
                    result[i].append(result[i - 1][j - 1] + result[i - 1][j])
        return result

File: C++/generatePascal.cpp
vector <vector<int>> Solution::generate(int A) {
    vector <vector<int>> ans;
    if (A <= 0) {
        return ans;
    }
    vector<int> temp;
    temp.push_back(1);
    ans.push_back(temp);

    for (int r = 0; r < A - 1; r++) {

        vector<int> newRow;
        newRow.push_back(1);

        for (int c = 0; c < ans[r].size() - 1; c++) {
            newRow.push_back(ans[r][c] + ans[r][c + 1]);
        }

        newRow.push_back(1);
        ans.push_back(newRow);

    }
    return ans;
}

